src/interactive-window/interactiveWindowProvider.ts

```diff
@@ -186,17 +184,11 @@ export class InteractiveWindowProvider
         return noop();
     }
 
-    // Note to future devs: this function must be synchronous. Do not await on anything before calling
-    // the interactive window ctor and adding the interactive window to the provider's list of known windows.
-    // Otherwise we risk a race condition where e.g. multiple run cell requests come in quick and we end up
-    // instantiating multiples.
     private async create(resource: Resource, mode: InteractiveWindowMode, connection?: KernelConnectionMetadata) {
+        // track when a creation is pending, so consumers can wait before checking for an existing one.
         const creationInProgress = createDeferred<void>();
-        // Ensure we don't end up calling this method multiple times when creating an IW for the same resource.
         this.pendingCreations.push(creationInProgress.promise);
```

Comment:
Reading `pendingCreation`, it gave me a feeling that we can have multiple creation in parallel. However we are awaiting for all pending creations to be complete before creating new ones, does it mean that `pendingCreation.length <= 1` always? If so we can probably change it to `pendingCreations: Promise<void> | null`

Reply:
yeah I noticed that as well, I'll change it to better express that only one creation is happening at a time